src/components/ViewerPageContainer.js

```diff
@@ -30,9 +30,9 @@ class MainPageContainer extends Component {
   }
 
   componentDidMount() {
-    const { document } = this.props;
+    const { document, history } = this.props;
     if (!document) {
-      Router.replace("/");
+      history.push("/");
```

Comment:
not exactly re-routing with this code. even before getting  "page not found" error.

Reply:
fixed in netlify.toml